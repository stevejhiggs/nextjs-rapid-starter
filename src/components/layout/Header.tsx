import Link from 'next/link';
import { useState } from 'react';
import { FireIcon, ShareIcon, MapPinIcon } from '@heroicons/react/24/outline';

const Header: React.FC = () => {
  const [navbarOpen, setNavbarOpen] = useState(false);

  return (
    <nav className="relative flex flex-wrap items-center justify-between px-2 py-3 bg-blue-500 mb-3">
      <div className="px-4 w-full mx-auto flex flex-wrap items-center justify-between">
        <div className="w-full relative flex justify-between lg:w-auto lg:static lg:block lg:justify-start">
          <Link
            data-testid="page-heading-title"
            className="text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase text-white"
            href="/"
          >
            Next.js starter
          </Link>
          <button
            className="text-white cursor-pointer text-xl leading-none px-3 py-1 border border-solid border-transparent rounded bg-transparent block lg:hidden outline-none focus:outline-none"
            type="button"
            onClick={() => setNavbarOpen(!navbarOpen)}
          >
            <i className="fas fa-bars">
              <svg fill="white" width="12px" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
                <title>Menu</title>
                <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
              </svg>
            </i>
          </button>
        </div>
        <div className={'lg:flex flex-grow items-center' + (navbarOpen ? ' flex' : ' hidden')} id="example-navbar-danger">
          <ul className="flex flex-col lg:flex-row list-none lg:ml-auto">
            <li className="nav-item">
              <a className="px-3 py-2 flex items-center text-xs uppercase font-bold leading-snug text-white hover:opacity-75" href="#pablo">
                <ShareIcon className="h-6 w-6 text-white" />
                <span className="ml-2">Share</span>
              </a>
            </li>
            <li className="nav-item">
              <a className="px-3 py-2 flex items-center text-xs uppercase font-bold leading-snug text-white hover:opacity-75" href="#pablo">
                <FireIcon className="h-6 w-6 text-white" />
                <span className="ml-2">Tweet</span>
              </a>
            </li>
            <li className="nav-item">
              <a className="px-3 py-2 flex items-center text-xs uppercase font-bold leading-snug text-white hover:opacity-75" href="#pablo">
                <MapPinIcon className="h-6 w-6 text-white" />
                <span className="ml-2">Pin</span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Header;
