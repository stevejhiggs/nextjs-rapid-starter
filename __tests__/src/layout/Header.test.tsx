import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import Header from '@/src/layout/Header';

afterEach(cleanup);

describe('Components -> Header', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('Should have the right content', () => {
    render(<Header />);
    const title = screen.getByTestId('page-heading-title');
    expect(title).toHaveTextContent('Next.js starter');
  });
});
