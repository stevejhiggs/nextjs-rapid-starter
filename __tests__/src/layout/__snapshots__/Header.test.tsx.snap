// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Header should match the snapshot 1`] = `
<div>
  <nav
    class="css-1hn0bm1"
  >
    <div
      class="css-1c714fi"
    >
      <h1
        class="chakra-heading css-0"
        data-testid="page-heading-title"
      >
        Next.js starter
      </h1>
    </div>
    <div
      class="css-m9ssb"
    >
      <svg
        fill="white"
        viewBox="0 0 20 20"
        width="12px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Menu
        </title>
        <path
          d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
        />
      </svg>
    </div>
    <div
      class="css-t2kg2r"
    >
      <p
        class="chakra-text css-dtinqn"
      >
        Menu1
      </p>
      <p
        class="chakra-text css-dtinqn"
      >
        Menu2
      </p>
      <p
        class="chakra-text css-dtinqn"
      >
        Menu3
      </p>
    </div>
    <div
      class="css-mg6i13"
    >
      <button
        class="chakra-button css-1tvtpuq"
        type="button"
      >
        Example Button
      </button>
    </div>
  </nav>
</div>
`;
